test/server/media-controller.spec.js

```diff
@@ -0,0 +1,150 @@
+const chai = require( 'chai' );
+const expect = chai.expect;
+const request = require( 'request' );
+const config = require( '../../app/models/config-model' ).server;
+
+const IPfiltering = config[ 'ip filtering' ];
+
+/**
+ * Tests the request-filtering-agent to block SSRF attacks
+ * change testHTMLBody to the body of an html file that
+ * you are testing on. For the default, it says <i'm in.>
+ * and is hosted in testHTMLHost. 
+ */
+
+const testHTMLBody = 'i\'m in.\n';
+const enketoHost = 'http://localhost:8005';
+const testHTMLHost = 'http/localhost:1234';
+
+const requestURL = enketoHost + '/media/get/' + testHTMLHost;
+
+// TODO: Check if testHTMLHost is running
+
+describe( 'Media Controller', () => {
+
+    console.log( 'Testing request-filtering-agent on this request: ' + requestURL);
+    
+    describe( 'No Referer Request', () => {
+
+        if (!IPfiltering.allowPrivateIPAddress) {
+            if (IPfiltering.allowIPAddressList === '' ) {
+                it( 'allowPrivateIPAddress is false', () => {
+                    request( requestURL ,
+                        function(error, response, body){
+                            expect(body).to.be.equal(undefined);
```

Comment:
A more explicit check for expect(error).to.include('SOMEERROR') would be better I think. It avoids the false positives, that I think you are getting currently, assuming that a useful descriptive SOMEERROR is returned by the filtering agent.